File: 1537-maximum-score-after-splitting-a-string/maximum-score-after-splitting-a-string.cpp
class Solution {
public:
    int maxScore(string s) {
        int ones = 0, zero = 0, ans = 0, sz = s.length();

        for(auto i : s)
            if(i == '1') ones++;

        for(int i=0; i<sz-1; i++){
            if(s[i] == '1') ones--;
            else zero++;
            ans = max(zero + ones, ans);
        }

        return ans;
    }
};

File: 1537-maximum-score-after-splitting-a-string/maximum-score-after-splitting-a-string.java
class Solution {
    public int maxScore(String s) {
        int sz = s.length(), ans = 0;
        int zero = 0, one = 0;

        for(int i=0; i<sz; i++){
            if(s.charAt(i) == '1') one++;
        }

        for(int i=0; i<sz-1; i++){
            if(s.charAt(i) == '0') zero++;
            else one--;
            ans = Math.max(ans, one+zero);
        }

        return ans;
    }
}